packages/components/src/DateFormat/helper.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Tooltip, TooltipProps } from '@patternfly/react-core';
+
+const second = 1000;
+const minute = second * 60;
+const hour = minute * 60;
+const day = hour * 24;
+const month = day * 30; // let's count that every month has 30 days
+const year = day * 365;
+const formatTime = (number: number, unit: string) => `${number} ${number > 1 ? `${unit}s` : unit} ago`;
+
+const relativeTimeTable = [
+  { rightBound: Infinity, description: (date: number) => formatTime(Math.round(date / year), 'year') },
+  { rightBound: year, description: (date: number) => formatTime(Math.round(date / month), 'month') },
+  { rightBound: month, description: (date: number) => formatTime(Math.round(date / day), 'day') },
+  { rightBound: day, description: (date: number) => formatTime(Math.round(date / hour), 'hour') },
+  { rightBound: hour, description: (date: number) => formatTime(Math.round(date / minute), 'minute') },
+  { rightBound: minute, description: () => 'Just now' },
+];
+
+const exact = (value: Date) => value.toUTCString().split(',')[1].slice(0, -7).trim();
+
+export const addTooltip = (date: React.ReactNode, element?: React.ReactElement, tooltipProps?: TooltipProps, extraTitle: React.ReactNode = '') => (
+  <Tooltip
+    {...tooltipProps}
+    content={
+      <div>
+        {extraTitle}
+        {date}
+      </div>
+    }
+  >
+    {element}
+  </Tooltip>
+);
+
+export type DateType = 'exact' | 'onlyDate' | 'relative' | 'invalid';
```

Comment:
Could we use this in DateFormatProps (with omitting the invalid), so the types are defined in one place?

Reply:
Exclude might work

Reply:
And it does